streams/src/test/java/org/apache/kafka/streams/state/internals/InMemoryKeyValueLoggedStoreTest.java

```diff
@@ -17,44 +17,197 @@
 import org.apache.kafka.streams.processor.StateStore;
 import org.apache.kafka.streams.state.KeyValueStore;
 import org.apache.kafka.streams.state.StoreBuilder;
 import org.apache.kafka.streams.state.Stores;
+import org.apache.kafka.test.KeyValueIteratorStub;
+import org.easymock.EasyMock;
+import org.junit.Before;
 import org.junit.Test;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
+import java.util.Iterator;
 import java.util.List;
 
-import static org.junit.Assert.assertEquals;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertThat;
 
 public class InMemoryKeyValueLoggedStoreTest extends AbstractKeyValueStoreTest {
 
+
+    private KeyValueStore<Integer, String> innerKeyValueStore = EasyMock.createNiceMock(KeyValueStore.class);
+    private InMemoryKeyValueLoggedStore<Integer, String> inMemoryKeyValueLoggedStore;
+
+    private final Serde<String> stringSerde = Serdes.String();
+    private final Serde<Integer> integerSerde = Serdes.Integer();
+
+
     @SuppressWarnings("unchecked")
     @Override
     protected <K, V> KeyValueStore<K, V> createKeyValueStore(final ProcessorContext context) {
         final StoreBuilder storeBuilder = Stores.keyValueStoreBuilder(
-                Stores.inMemoryKeyValueStore("my-store"),
-                (Serde<K>) context.keySerde(),
-                (Serde<V>) context.valueSerde())
-                .withLoggingEnabled(Collections.singletonMap("retention.ms", "1000"));
+            Stores.inMemoryKeyValueStore("my-store"),
+            (Serde<K>) context.keySerde(),
+            (Serde<V>) context.valueSerde())
+            .withLoggingEnabled(Collections.singletonMap("retention.ms", "1000"));
 
         final StateStore store = storeBuilder.build();
         store.init(context, store);
 
         return (KeyValueStore<K, V>) store;
     }
 
+    @Before
+    public void setUp() {
+        innerKeyValueStore.init(EasyMock.isA(ProcessorContext.class), EasyMock.isA(StateStore.class));
+        EasyMock.expect(innerKeyValueStore.name()).andReturn("inner-store").times(2);
+        inMemoryKeyValueLoggedStore = new InMemoryKeyValueLoggedStore<>(innerKeyValueStore, integerSerde, stringSerde);
+    }
+
     @Test
     public void shouldPutAll() {
         List<KeyValue<Integer, String>> entries = new ArrayList<>();
         entries.add(new KeyValue<>(1, "1"));
         entries.add(new KeyValue<>(2, "2"));
-        store.putAll(entries);
-        assertEquals(store.get(1), "1");
-        assertEquals(store.get(2), "2");
+
+        innerKeyValueStore.putAll(entries);
+        EasyMock.replay(innerKeyValueStore);
+
+        inMemoryKeyValueLoggedStore.init(context, innerKeyValueStore);
+        inMemoryKeyValueLoggedStore.putAll(entries);
+        EasyMock.verify(innerKeyValueStore);
+    }
+
+    @Test
+    public void shouldReturnCorrectApproximateNumberOfEntries() {
```

Comment:
Could the following two tests be in the parent `AbstractKeyValueStoreTest`?

Reply:
already exists in the parent, removing